articles/events_executor.md

```diff
@@ -129,11 +132,63 @@ This class will forward the items needed for pushing events to the notify guard
 Whenever an entity is associated to an executor it may already have multiple items of work ready to be processed.
 The `EventsExecutor` should tell the entity whether it has to immediately push or to discard those pending events.
 
+### Types of Events queue
+
+There are multiple uses cases with different requirements (performance, determinism in events ordering, bounded memory). A single `EventsQueue` can't comply with all requirements simultaneously, so there's need for different types of queue.
+
+##### SimpleEventsQueue
+
+This is the simplest queue, as it does not peform any checks when events are pushed or extracted, so it's more suitable for situacions where CPU performance is needed.
+
+As long as events are pushed into the queue it will keep growing, regardless of the actual history size of the underlying middleware entity.
+This may cause several events to accumulate, for example while the `EventsExecutor` is not spinning, which can lead to out-of-memory situations or subverting the ordering of events.
+
+This queue may fail to provide a correct ordering of events in some corner case situations.
+In particular, if an entity pushes a number of events greater than its QoS history size while the `EventsExecutor` is busy processing events, then the ordering may be compromised.
+
+Consider the following example of a system with two subscriptions A and B. Subscription A has a QoS history size of 1.
+While the `EventsExecutor` is busy processing events, the following events accumulates into the `EventsQueue`:
+ - Event 1 from Subscription A
+ - Event 2 from Subscription B
+ - Event 3 from Subscription A
+
+When the `EventsExecutor` is ready to process these new events, it will start from the first pushed, i.e. the event 1 from Subscription A.
+However, when calling the `execute_subscription()` API, the message that generated event 1 will not be available anymore as it has been overwritten from the message that generated event 3.
+The `execute_subscription()` API will not fail, but rather it will process the message that generated event 3.
+This violates the ordering as event 2 should have been processed before that.
+
+##### LockFreeEventsQueue
```

Comment:
let's remove this section and we can just mention above that if needed you can implement a lock free queue

Reply:
Removed and menitioned in 0d70bf2